web/profiles/custom/os2loop/modules/os2loop_analytics/src/Helper/AnalyticsHelper.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Drupal\os2loop_analytics\Helper;
+
+use Drupal\os2loop_analytics\Form\SettingsForm;
+use Drupal\os2loop_settings\Settings;
+use Drupal\Core\Render\Markup;
+
+/**
+ * Helper for os2loop_analytics.
+ */
+class AnalyticsHelper {
+  /**
+   * The config.
+   *
+   * @var \Drupal\Core\Config\ImmutableConfig
+   */
+  private $config;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\os2loop_settings\Settings $settings
+   *   Settings.
+   */
+  public function __construct(Settings $settings) {
+    $this->config = $settings->getConfig(SettingsForm::SETTINGS_NAME);
+  }
+
+  /**
+   * Set analytics tools script in head.
+   *
+   * @param array $attachments
+   *   The attachments to add.
+   */
+  public function setAnalytics(array &$attachments) {
+    $scripts = $this->getAnalyticsConfig();
+    foreach ($scripts as $name => $script) {
+      $cleanScript = $this->cleanup($script);
+      $attachments['#attached']['html_head'][] = [
+        [
+          '#tag' => 'script',
+          '#value' => Markup::create($cleanScript),
+          '#weight' => 1,
+        ],
+        $name . '_script',
+      ];
```

Comment:
I appreciate what you're doing here, but we may run into issues with tracking codes that require more that a single `script` element (e.g. https://developers.google.com/analytics/devguides/collection/analyticsjs#alternative_async_tag). We should consider just letting any code through without any modifications.

Reply:
Removed helper all together